javalib/src/main/scala/java/net/PlainSocketImpl.scala

```diff
@@ -269,16 +270,16 @@ private[net] class PlainSocketImpl extends SocketImpl {
 
     throwIfClosed(fd.fd, "connect") // Do not send negative fd.fd to poll()
 
-    val inetAddr = address.asInstanceOf[InetSocketAddress]
-    val hints    = stackalloc[addrinfo]
-    val ret      = stackalloc[Ptr[addrinfo]]
-    string.memset(hints.asInstanceOf[Ptr[Byte]], 0, sizeof[addrinfo])
-    hints.ai_family = socket.AF_UNSPEC
-    hints.ai_flags = AI_NUMERICHOST | AI_NUMERICSERV
-    hints.ai_socktype = socket.SOCK_STREAM
+    val inetAddr      = address.asInstanceOf[InetSocketAddress]
+    val ret           = stackalloc[Ptr[addrinfo]]
     val remoteAddress = inetAddr.getAddress.getHostAddress()
 
     Zone { implicit z =>
+      val hints = alloc[addrinfo] // alloc is documented to clear memory
+      hints.ai_family = socket.AF_UNSPEC
+      hints.ai_flags = AI_NUMERICHOST | AI_NUMERICSERV
+      hints.ai_socktype = socket.SOCK_STREAM
```

Comment:
That way these can be left out of the Zone. And also in the earlier spot.

Reply:
I like the locality of usage.  Agreed, they do not have to be in the Zone but neither
are harmed, nor do harm there. 

I checked if I could do away with the Zone entirely, but it is necessary for 
 some CString  conversion there.

Reply:
Right, but it is extra change that is not needed and it sort of implies that it is needed in the Zone which may mislead future devos.